Sources/Plasma/FeatureLib/pfPython/pyImageGlue.cpp

```diff
@@ -165,21 +165,35 @@ PYTHON_METHOD_DEFINITION(ptImage, saveAsJPEG, args)
 
 PYTHON_METHOD_DEFINITION(ptImage, saveAsPNG, args)
 {
-    PyObject* filenameObj;
-    if (!PyArg_ParseTuple(args, "O", &filenameObj))
+    PyObject* filenameObj, *optionalText;
+    if (!PyArg_ParseTuple(args, "O|O", &filenameObj, &optionalText))
     {
-        PyErr_SetString(PyExc_TypeError, "saveAsPNG expects a string");
+        PyErr_SetString(PyExc_TypeError, "saveAsPNG expects a string, and an optional dict of key-value pairs");
         PYTHON_RETURN_ERROR;
     }
 
     if (PyString_CheckEx(filenameObj))
     {
-        self->fThis->SaveAsPNG(PyString_AsStringEx(filenameObj));
+        if (PyDict_Check(optionalText)) {
+            auto textFields = new std::multimap<plString, plString>();
```

Comment:
Considering that this is a temporary object that does not persist outside this block, can we allocate the multimap on the stack?